File: snippets/rust/src/buy_btc.rs
use std::sync::Arc;

use anyhow::Result;
use breez_sdk_core::*;

async fn buy(sdk: Arc<BreezServices>) -> Result<()> {
    // ANCHOR: buy-btc
    let res = sdk.buy_bitcoin(
        BuyBitcoinRequest {
            provider: BuyBitcoinProvider::Moonpay,
            opening_fee_params: None})
        .await?;
    // ANCHOR_END: buy-btc

    Ok(())
}

File: snippets/react-native/buy_btc.ts
import {
  buyBitcoin,
  BuyBitcoinProvider
} from '@breeztech/react-native-breez-sdk'

const exampleBuyBtc = async () => {
  // ANCHOR: buy-btc
  try {
    const buyBitcoinResponse = await buyBitcoin({
      provider: BuyBitcoinProvider.MOONPAY
    })
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: buy-btc
}
